tests/python/pants_test/backend/jvm/tasks/jvm_compile/test_jvm_compile.py

```diff
@@ -42,10 +58,43 @@ def test_if_runtime_classpath_exists(self):
     runtime_classpath.add_for_targets([target], [('default', pre_init_runtime_entry)])
 
     task = self.create_task(context)
-    resulting_classpath = task.create_runtime_classpath()
+    resulting_classpath = task.create_classpath_product()
     self.assertEqual([('default', pre_init_runtime_entry), ('default', compile_entry)],
       resulting_classpath.get_for_target(target))
 
+  def test_modulized_targets_not_compiled_for_export_classpath(self):
+    init_subsystems([JvmPlatform])
+    targets = self.make_linear_graph(['a', 'b', 'c', 'd', 'e'], target_type=JavaLibrary)
+    context = self.context(target_roots=[targets['a'], targets['c']], options={'jvm-platform': {'compiler': 'dummy'}})
+    context.products.get_data('compile_classpath', ClasspathProducts.init_func(self.pants_workdir))
+    # This should cause the jvm compile execution to exclude target roots and their
+    # dependess from the set of relevant targets.
+    context.products.require_data("export_dep_as_jar_classpath")
+    self.execute(context)
+    export_classpath = context.products.get_data('export_dep_as_jar_classpath')
```

Comment:
This on the surface looks like it could be extracted a method that takes in the requested products and return the classpath_product, to share with the other test.

Reply:
good point. There is definitely overlap.